src/components/EditCategoryModal/EditCategoryModal.jsx

```diff
@@ -0,0 +1,180 @@
+import { useEffect } from 'react';
+import { Controller, useForm } from 'react-hook-form';
+import { useQuery } from 'react-query';
+import {
+  Button,
+  Flex,
+  FormControl,
+  FormErrorMessage,
+  FormLabel,
+  Input,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Text,
+  useDisclosure
+} from '@chakra-ui/react';
+import { ColorPicker } from 'chakra-color-picker';
+import i18next from 'i18next';
+
+import { getCategories } from '@/api/Category';
+import { ConfirmationModal } from '@/components';
+
+export const EditCategoryModal = ({
+  isOpen,
+  onClose,
+  onSubmit,
+  categoryType,
+  categoryData
+}) => {
+  const { data: dataCategories, isFetched: isFetchedCategories } = useQuery(
+    ['categories'],
+    getCategories
+  );
+
+  const categoriesDeleteModal = useDisclosure();
+
+  const {
+    register,
+    handleSubmit,
+    control,
+    reset,
+    isDirty,
+    formState: {
+      errors: { name }
+    }
+  } = useForm({
+    defaultValues: {
+      categoryType: categoryData.categoryType,
+      categoryId: categoryData.id,
+      name: categoryData.name,
+      color: categoryData.color
+    }
+  });
+  const resetForm = () => {
+    reset({
+      categoryType: categoryData.categoryType,
+      categoryId: categoryData.id,
+      name: categoryData.name,
+      color: categoryData.color
+    });
+  };
+  useEffect(() => resetForm(), [!isOpen]);
+
+  const closeAllModals = () => {
+    categoriesDeleteModal.onClose();
+    onClose();
+  };
+  //validation does not work, because the form already has default fields, and when compared in isDirty, it considers that everything has been changed
```

Comment:
can we do fix for that?

Reply:
I fix it! and I fix it in AddCategoryModal to!